File: foresight-api/src/main/java/br/com/foresight/modules/comercial/produto/dto/ProdutoDto.java
package br.com.foresight.modules.comercial.produto.dto;

import java.math.BigDecimal;

public record ProdutoDto(
        Long id,
        String nome,
        String codigoBarras,
        String imagemUrl,
        Long categoriaId,
        String categoriaNome,
        Long fornecedorId,
        String fornecedorNome,
        BigDecimal precoCusto,
        BigDecimal precoVenda,
        Integer estoqueAtual,
        Integer estoqueMinimo,
        BigDecimal margemReal,
        BigDecimal lucroReal,
        String alertaStatus,
        // NOVOS CAMPOS PARA FINANCIAMENTO
        Long investidorId,
        String investidorNome,
        BigDecimal percentualLucroInvestidor
) {}

File: foresight-web/src/app/features/produtos/models/produto.dto.ts
export interface ProdutoRequest {
  nome: string;
  codigoBarras?: string | null;
  imagemUrl?: string | null;
  categoriaId?: number | null;
  fornecedorId?: number | null;
  precoCusto: number;
  precoVenda: number;
  estoqueAtual: number;
  estoqueMinimo: number;
  investidorId?: number | null;
  percentualLucroInvestidor?: number | null;
}

export interface ProdutoDto {
  id: number;
  nome: string;
  codigoBarras?: string | null;
  imagemUrl?: string | null;
  categoriaId?: number | null;
  categoriaNome?: string | null;
  fornecedorId?: number | null;
  fornecedorNome?: string | null;
  precoCusto: number;
  precoVenda: number;
  estoqueAtual: number;
  estoqueMinimo: number;
  lucroReal: number;
  margemReal: number;
  alertaStatus: string;
}
